socs/agents/acu/agent.py

```diff
@@ -2067,22 +2072,24 @@ def _run_track(self, session, point_gen, step_time, azonly=False,
                     # after "grabbing the minimum batch", below, there
                     # is still >= 1 line left.  The lines-is-empty
                     # check is used to decide we're done.
-                    while mode == 'go' and (len(lines) <= new_line_target or lines[-1][0] != 0):
+                    while mode == 'go' and (len(lines) <= new_line_target or lines[-1].group_flag != 0):
                         try:
                             lines.extend(next(point_gen))
                         except StopIteration:
                             mode = 'stop'
+                            stop_message = 'Stop due to end of the planned track.'
 
                     # Grab the minimum batch
                     upload_lines, lines = lines[:new_line_target], lines[new_line_target:]
 
                     # If the last line has a "group" flag, keep transferring lines.
-                    while len(lines) and len(upload_lines) and upload_lines[-1][0] != 0:
+                    while len(lines) and len(upload_lines) and upload_lines[-1].group_flag != 0:
                         upload_lines.append(lines.pop(0))
 
                     if len(upload_lines):
                         # Discard the group flag and upload all.
-                        text = ''.join([line for _flag, line in upload_lines])
+                        text = sh.get_track_points_text(
+                            upload_lines, timestamp_offset=3, text_block=True)
```

Comment:
Just trying to make sure I follow. This is what's breaking `fromfile_scan`? Because `_run_track` now expects a list of `PointTrack` objects, rather than the already formatted list of text?

When we merge this, would you mind opening an issue about the broken `fromfile_scan` with some details?